rust/gui-client/docs/process_split.md

```diff
@@ -0,0 +1,39 @@
+# Process split
+
+This is meant for Linux, but it will probably be very similar on Windows.
+
+## TODO
+
+- Do we need to support CLI and GUI Clients being installed on the same system? The tunnel binary would conflict if they both try to install it. Splitting it into 3 packages would complicate downloads.
+- Clients and Gateways can't run in the same system, right? How would DNS work if they did?
+- If there is a service account token, but it's invalid, should the tunnel wait for the GUI to connect, or fail? Will the tunnel know whether a GUI is even installed?
+
+## Binaries
+
+I gave them new names to clarify.
+
+1. `firezoned` - A daemon that runs the tunnel and has elevated privilege
```

Comment:
I was even thinking to split into `firezone-cli-client` and `firezone-gui-client` so that if someone tries to run the GUI in systemd it'll show a more obvious error. It's easier to make names shorter later on than to split up conflicting names after they've gone into docs and config files